test/functional/rpc_getblockstats.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test getblockstats rpc call
+#
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+def assert_contains(data, values, check_cointains=True):
+    for val in values:
+        if check_cointains:
+            assert val in data
+        else:
+            assert val not in data
+
+class GetblockstatsTest(BitcoinTestFramework):
+    EXPECTED_STATS = [
+        "height",
+        "time",
+        "mediantime",
+        "txs",
+        "swtxs",
+        "ins",
+        "outs",
+        "subsidy",
+        "totalfee",
+        "utxo_increase",
+        "utxo_size_inc",
+        "total_size",
+        "total_weight",
+        "swtotal_size",
+        "swtotal_weight",
+        "total_out",
+        "minfee",
+        "maxfee",
+        "medianfee",
+        "avgfee",
+        "minfeerate",
+        "maxfeerate",
+        "medianfeerate",
+        "avgfeerate",
+        "mintxsize",
+        "maxtxsize",
+        "mediantxsize",
+        "avgtxsize",
+    ]
+
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.extra_args = [['-txindex'], ['-paytxfee=0.003']]
```

Comment:
commit: Tests: Test new getblockstats RPC

I think it's worth testing the case of no txindex.